include/openthread/border_router.h

```diff
@@ -117,6 +117,15 @@ otError otBorderRoutingGetOmrPrefix(otInstance *aInstance, otIp6Prefix *aPrefix)
  */
 otError otBorderRoutingGetOnLinkPrefix(otInstance *aInstance, otIp6Prefix *aPrefix);
 
+/**
+ * This method checks if the On Mesh Prefix is a valid off-mesh-routable (OMR) Prefix.
+ *
+ * @param[in]  aConfig    A pointer the the On Mesh Prefix inforamtion.
+ *
+ * @returns  Whether the On Mesh Prefix is a valid OMR Prefix.
+ */
+bool otBorderRoutingIsValidOmrPrefix(const otBorderRouterConfig *aConfig);
```

Comment:
It calls `BorderRouter::RoutingManager::IsValidOmrPrefix`, so I think `otBorderRoutingIsValidOmrPrefix` is proper. 

It simply checks if the prefix config is a legit OMR prefix.

it does not check the prefix in the Network Data, so `otNetDataIsValidOmrPrefix` might be misleading?

Reply:
https://github.com/openthread/openthread/blob/1091b947dc1e1c41fd3c7fc5a7d45c58dfa16666/src/core/border_router/routing_manager.cpp#L954-L964

If looking into the implementation, `IsValidOmrPrefix(const NetworkData::OnMeshPrefixConfig &aOnMeshPrefixConfig)` or `IsValidOmrPrefix(const Ip6::Prefix &aOmrPrefix)` should not be closely tied to routing manager in my view.

---
Anyway, if there are common `otIp6IsOmrAddress` and `otNetDataHasOmrPrefix` which checks valid omr prefix using internal methods, it might be not necessary to have this API as far as now is concerned.

Reply:
`otNetDataContainsOmrPrefix` added in the last commit. 

`otIp6IsOmrAddress` is not added because it can be easily implemented by `otNetDataContainsOmrPrefix`.